### PR TITLE
Budget features

### PR DESCRIPTION
**Pull request by DianaBeki**
### Project requirements
[Day 1](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#day-1)

-  Set up the repository and tools.
- Create your models according to the ERD diagram. Don't forget about validations and tests.
- Implement the authentication and authorization.

[Day 2](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#day-2)

- Create the page to add a category.
- Create the home page.
- Create the page to add a transaction.
- Create the transactions page.

[Day 3](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#day-3)

- Make sure your app is tested adequately.
- Deploy the project, and test for final details.
- Record a video for your project.
- Create a good README and PR description

### CODE REVIEW

File: spec/models/category_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+  it 'should contain a name of category' do
+    category.name = nil
+    expect(category).to_not be_valid
+  end
+
+  it 'should contain an icon of category' do
+    category.icon = nil
+    expect(category).to_not be_valid
+  end
```
Comment (OLIPLICHE) [inline spec/models/category_spec.rb:14]: - [ ] Hello, nice job so far. I suggest you work on the Testing again because the test did not pass.

Capture

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @DianaBeki, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Build a mobile web application 👍 
- Able to make a list of transactions associated with a category 👍
-  The user can log in to the app using email and password 👍 
- Use Postgres as the database 👍 
- use devise for authentication 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/category_spec.rb (hunk 1 above)
Comment (DianaBeki) [reply spec/models/category_spec.rb:14]: @OLIPLICHE seen the changes but on my terminal I made sure the tests were passing for me to be able to submit and here is the evidence cos I did it  again now .Kindly confirm in your side .Thank you
spec

**Comment by OLIPLICHE** [comment]
- [ ]  Hi, **This is for the second reviewer**, I met @DianaBeki on Slack who just showed me that the test on Categorie file has passed at her end. I humbly asked the second reviewer to check it again and confirm and approve the project if it works. Thank you.

**Review by prolajumokeoni — APPROVED** [review]
Hi @DianaBeki 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - Resolved requested change with reviewer ✅ 
 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**